libs/filer/workspace_files_client.go

```diff
@@ -137,11 +137,22 @@ func (w *WorkspaceFilesClient) Write(ctx context.Context, name string, reader io
 		// Create parent directory.
 		err = w.workspaceClient.Workspace.MkdirsByPath(ctx, path.Dir(absPath))
 		if err != nil {
+			if errors.As(err, &aerr) && aerr.StatusCode == http.StatusForbidden {
+				return PermissionError{absPath}
+			}
 			return fmt.Errorf("unable to mkdir to write file %s: %w", absPath, err)
 		}
 
 		// Retry without CreateParentDirectories mode flag.
-		return w.Write(ctx, name, bytes.NewReader(body), sliceWithout(mode, CreateParentDirectories)...)
+		err = w.Write(ctx, name, bytes.NewReader(body), sliceWithout(mode, CreateParentDirectories)...)
+		notExistsError := NoSuchDirectoryError{}
+		if errors.As(err, &notExistsError) {
+			// If we still get a 404 error after the succesful mkdir,
+			// the problem is a permission error.
+			return PermissionError{absPath}
```

Comment:
Did you observe this?

Reply:
Yeah. This is the case where you the dir exists and you have 'view' access but not 'write' access. We previously couldn't recognize this case as a permission error.

Reply:
We need integration test coverage for this.

Let's figure out a way to do so, because we don't have any tests that use multiple identities at the moment, and we'll need that in order to test this branch (setup the tree to trigger this path).

Same holds for the permission checks in the rest of this PR. Without integration test coverage we can't guarantee continuity, or even that the assertions made continue to hold.

Reply:
Sounds good, let's discuss offline what kind of coverage / testing approach would be coverage.

Reply:
Updated as discussed, removing the new filer logic for now: https://github.com/databricks/cli/pull/1386/commits/26c3b421e84542e1fc59d292eb1f936528276010